packages/react-native-gesture-handler/src/v3/components/GestureComponents.tsx

```diff
@@ -0,0 +1,195 @@
+import React, {
+  PropsWithChildren,
+  ReactElement,
+  useRef,
+  useImperativeHandle,
+  useState,
+  RefObject,
+} from 'react';
+import {
+  ScrollView as RNScrollView,
+  ScrollViewProps as RNScrollViewProps,
+  Switch as RNSwitch,
+  SwitchProps as RNSwitchProps,
+  TextInput as RNTextInput,
+  TextInputProps as RNTextInputProps,
+  FlatList as RNFlatList,
+  FlatListProps as RNFlatListProps,
+  RefreshControl as RNRefreshControl,
+} from 'react-native';
+
+import createNativeWrapper, {
+  ComponentWrapperRef,
+} from '../createNativeWrapper';
+
+import { NativeWrapperProperties } from '../types/NativeWrapperType';
+import { NativeWrapperProps } from '../hooks/utils';
+import { DetectorType } from '../detectors';
+import { NativeGesture } from '../hooks/gestures/native/useNativeGesture';
+
+export const RefreshControl = createNativeWrapper(
+  RNRefreshControl,
+  {
+    disallowInterruption: true,
+    shouldCancelWhenOutside: false,
+  },
+  DetectorType.Virtual
+);
+
+// eslint-disable-next-line @typescript-eslint/no-redeclare
+export type RefreshControl = typeof RefreshControl & RNRefreshControl;
+
+const GHScrollView = createNativeWrapper<PropsWithChildren<RNScrollViewProps>>(
+  RNScrollView,
+  {
+    disallowInterruption: true,
+    shouldCancelWhenOutside: false,
+  },
+  DetectorType.Intercepting
+);
```

Comment:
How does this work when, correct me if I'm wrong, Android and iOS render them in different positions, e.g.:

```
<RefreshControl>
  <ScrollView />
</RefreshControl>
```

on one and 

```
<ScrollView>
  <RefreshControl />
</ScrollView>
```

on the other. I'd imagine that would also impact where the detector is rendered.